Parsers/ASimFileEvent/Parsers/vimFileEventVMwareCarbonBlackCloud.yaml

```diff
@@ -0,0 +1,150 @@
+      "ACTION_FILE_SET_SECURITY", "FileAttributesUpdated",
+      "ACTION_FILE_TRUNCATE", "FileModified",
+      "ACTION_FILE_OPEN_WRITE", "FileModified",
+      "ACTION_FILE_MOD_OPEN", "FileAccessed",
+      "ACTION_FILE_OPEN_READ", "FileAccessed"
+  ];
+  let parser = (disabled: bool=false) {
+      CarbonBlackEvents_CL
+      | where eventType_s == "endpoint.event.filemod" and isnotempty(filemod_name_s)
+      | where action_s !in ("ACTION_INVALID", "ACTION_FILE_UNDELETE")
+      | parse filemod_hash_s with * '["' TargetFileMD5: string '","' TargetFileSHA256: string '"]'
+      | lookup EventFieldsLookup on sensor_action_s
+      | extend temp_action = iff(action_s has "|", action_s, "")
+      | lookup EventTypeLookup on action_s
+      | extend EventType = case(
+                          isnotempty(EventType), EventType,
+                          temp_action has "delete", "FileDeleted",
+                          temp_action has "link", "FileModified",
+                          temp_action has "rename", "FileRenamed",
+                          temp_action has "execute", "FileAccessed",
+                          temp_action has_any ("attributes", "security"), "FileAttributesUpdated",
+                          temp_action has "truncate", "FileModified",
+                          temp_action has "write", "FileModified",
+                          temp_action has_any ("read", "open"), "FileAccessed",
+                          temp_action has "create", "FileCreated",
+                          ""
+                      )
+      | extend
+          EventStartTime = todatetime(split(createTime_s, '+')[0]),
+          TargetFilePathType = case(
+                          device_os_s == "WINDOWS" and filemod_name_s startswith "\\", "Windows Share",
+                          device_os_s == "WINDOWS", "Windows Local",
+                          device_os_s in ("MAC", "LINUX"), "Unix",
+                          ""
+                      ),
+          ActingProcessId = tostring(toint(process_pid_d)),
+          TargetFileName = tostring(split(filemod_name_s, '\\')[-1]),
+          AdditionalFields = bag_pack(
+                        "org_key", org_key_s,
+                        "process_publisher", process_publisher_s,
+                        "process_reputation", process_reputation_s,
+                        "process_guid", process_guid_s
+                    )
+      | invoke _ASIM_ResolveDvcFQDN('device_name_s')
+      | project-rename
+          ActorUsername = process_username_s,
+          DvcIpAddr = device_external_ip_s,
+          EventUid = _ItemId,
+          DvcScope = device_group_s,
+          ActingProcessCommandLine = process_cmdline_s,
+          ActingProcessName = process_path_s,
+          DvcId = device_id_s,
+          DvcOriginalAction = sensor_action_s,
+          DvcOs = device_os_s,
+          EventMessage = event_description_s,
+          EventOriginalType = action_s,
```

Comment:
what does this long string mean? Can we trim and store the required value in source format?

Reply:
![image](https://github.com/Azure/Azure-Sentinel/assets/97222872/45559395-82c7-4870-99b3-b5d78dc319e1)

Reply:
We are getting these values from the source itself and the action_s field indicates the operation associated with file. Hence we have added them in EventOriginalType.